lib/validators/csp_script_src_validator.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class CspScriptSrcValidator
+  def initialize(opts = {})
+    @opts = opts
+  end
+
+  def valid_value?(values)
+    return true if values == ""
+
+    regex =
+      /
+        (?:\A'unsafe-eval'\z)|
+        (?:\A'wasm-unsafe-eval'\z)|
+        (?:\A'sha(?:256|384|512)-[A-Za-z0-9+\/\-_]+={0,2}'\z)
+      /x
+
+    values.split("|").all? { |v| v.match? regex }
```

Comment:
More of a FYI, we could use numbered parameters here: `all? { _1.match?(regex) }`